src/main/lib/course_portfolio.js

```diff
@@ -73,4 +73,35 @@ module.exports.get = async (portfolio_id) => {
     }
 
     return portfolio
+}
+
+module.exports.getRandomStudents = async (students) => {
+    var num_students = 0
+    if(students.length === 0){
+        throw new Error('The must be students.')
+    }
+    if(students.length <= 10){
+        return students
+    }
+    else if(Math.floor(0.2 * students.length) < 10){
+        num_students = 10
+    }
+    else{
+        num_students = Math.floor(0.2 * students.length)
+    }
+    var random_students = []
+    var indices = []
+    var valid_index
+    for(var i = 0; i < num_students; i++){
```

Comment:
This may be more readable with some additional comments, or perhaps randomly sorting then pulling the first n entries